lib/webauthn/public_key_credential_with_assertion.rb

```diff
@@ -31,5 +32,20 @@ def user_handle
     def raw_user_handle
       response.user_handle
     end
+
+    private
+
+    def appid_extension_output
+      return if client_extension_outputs.nil?
+
+      client_extension_outputs[:appid] || client_extension_outputs['appid']
+    end
+
+    def appid
+      appid_domain = relying_party.legacy_u2f_appid || relying_party.id ||
+                     relying_party.origin || raise("Unspecified expected origin")
+
+      URI.parse(appid_domain).to_s
+    end
```

Comment:
Correct me if I'm wrong but I suspect this is incorrect – we should only check for the `legacy_u2f_appid` if the `appid` extension is present given that if it's not present, even if we pass in the `rp_id` the verification will fail. 

I would also consider raising an exception if the `appid` extension is present but the `legacy_u2f_appid` was not set 🤔

Reply:
Updated in [4058bea](https://github.com/cedarcode/webauthn-ruby/pull/319/commits/4058bea40a556a79c468d350bdb48e4419bce53d).

Reply:
> we should only check for the `legacy_u2f_appid` if the `appid` extension is present given that if it is not set, even if we pass in the `rp_id`, the verification will fail.

I confirmed that this is the case: if I leave `PublicKeyCredentialWithAssertion#appid` to be:

```ruby
    def appid
      appid_domain = relying_party.legacy_u2f_appid || relying_party.id ||
                     relying_party.origin || raise("Unspecified expected origin")

      URI.parse(appid_domain).to_s
    end
```

then [this spec that I created](https://github.com/cedarcode/webauthn-ruby/blob/4beef45aeb0e7a0d5cc5b8b837ae1145cf21dcfc/spec/webauthn/public_key_credential_with_assertion_spec.rb#L312:L320) will fail with a `WebAuthn::RpIdVerificationError`